File: test/lit/ts/shifts.ts
// RUN: %cot emit-cir %s | %FileCheck %s

// Phase 1: shift operators

function shl(a: number, b: number): number {
    return a << b;
}

function shr(a: number, b: number): number {
    return a >> b;
}

// CHECK-LABEL: func.func @shl
// CHECK: cir.shl

// CHECK-LABEL: func.func @shr
// CHECK: cir.shr

File: test/lit/swift/shifts.swift
// RUN: %cot emit-cir %s | %FileCheck %s

// libsc: shift operators << and >>

func shiftLeft(x: Int32, n: Int32) -> Int32 {
    return x << n
}

func shiftRight(x: Int32, n: Int32) -> Int32 {
    return x >> n
}

// CHECK-LABEL: func.func @shiftLeft
// CHECK: cir.shl %arg0, %arg1 : i32
// CHECK: return

// CHECK-LABEL: func.func @shiftRight
// CHECK: cir.shr %arg0, %arg1 : i32
// CHECK: return
